### PR TITLE
split FutureResult to interface and implementation

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

File: src/main/java/com/gooddata/FutureResultImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+package com.gooddata;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.gooddata.util.Validate.notNull;
+
+/**
+ * Represents the result retrieved by polling on the REST API.
+ */
+public final class FutureResultImpl<T> implements FutureResult<T> {
```
Comment (martiner) [inline src/main/java/com/gooddata/FutureResultImpl.java]: looks fine, but please some pick a better name

Comment (sadam21) [inline src/main/java/com/gooddata/FutureResultImpl.java]: For example `PollResult` ?

Comment (martiner) [inline src/main/java/com/gooddata/FutureResultImpl.java]: sure

Comment (sadam21) [inline src/main/java/com/gooddata/FutureResultImpl.java]: done

File: src/main/java/com/gooddata/FutureResult.java [hunk 2]
```diff
@@ -5,28 +5,10 @@
 
 import java.util.concurrent.TimeUnit;
 
-import static com.gooddata.util.Validate.notNull;
-
 /**
- * Represents the result retrieved by polling on the REST API.
```
Comment (martiner) [inline src/main/java/com/gooddata/FutureResult.java]: why such change?

**Comment by martiner** [comment]
Closes: #144

**Pull request merged**